modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/definitions/portlet/action/GetObjectFieldDeleteInfoMVCResourceCommand.java

```diff
@@ -50,34 +56,82 @@ protected void doServeResource(
 		JSONPortletResponseUtil.writeJSON(
 			resourceRequest, resourceResponse,
 			JSONUtil.put(
+				"deleteLastPublishedObjectDefinitionObjectField",
+				() -> _shouldDeleteLastPublishedObjectDefinitionObjectField(
+					objectDefinition, objectField)
+			).put(
+				"deleteObjectFieldObjectValidationRuleSetting",
+				() -> _shouldDeleteObjectFieldObjectValidationRuleSetting(
+					objectDefinition, objectField)
+			).put(
 				"showDeletionModal",
-				() -> {
-					if (!objectDefinition.isApproved()) {
-						return false;
-					}
+				() -> _shouldShowDeletionModal(objectDefinition, objectField)
+			));
+	}
 
-					return true;
-				}
-			).put(
-				"showDeletionNotAllowedModal",
-				() -> {
-					if (!objectDefinition.isApproved() ||
-						objectDefinition.isSystem()) {
+	private boolean _shouldDeleteLastPublishedObjectDefinitionObjectField(
+		ObjectDefinition objectDefinition, ObjectField objectField) {
 
-						return false;
-					}
+		if (!objectDefinition.isApproved() || objectDefinition.isSystem()) {
+			return true;
+		}
+
+		int customObjectFieldsCount =
+			_objectFieldLocalService.getObjectFieldsCount(
+				objectField.getObjectDefinitionId(), false);
+
+		if (customObjectFieldsCount <= 1) {
+			return false;
+		}
+
+		return true;
+	}
+
+	private boolean _shouldDeleteObjectFieldObjectValidationRuleSetting(
+		ObjectDefinition objectDefinition, ObjectField objectField) {
 
-					int customObjectFieldsCount =
-						_objectFieldLocalService.getObjectFieldsCount(
-							objectField.getObjectDefinitionId(), false);
+		for (ObjectValidationRule objectValidationRule :
+				_objectValidationRuleLocalService.
+					getObjectValidationRulesByObjectValidationRuleEngine(
+						objectDefinition.getObjectDefinitionId(),
```

Comment:
@igor-franca we can use the `getObjectDefinitionId()` from the `objectField` to simplify, i.e.:

```
private boolean _shouldDeleteObjectFieldObjectValidationRuleSetting(
	ObjectField objectField) {

	for (ObjectValidationRule objectValidationRule :
			_objectValidationRuleLocalService.getObjectValidationRules(
				objectField.getObjectDefinitionId(),
```